File: Harish Kumar/Trees/Binary Tree Level Order Traversal/Binary Tree Level Order Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<List<Integer>> ans = new ArrayList<>();
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        Queue<TreeNode> q = new LinkedList<>();
        
        if(root == null)
        {
            return ans;
        }
        
        q.offer(root);
        
        while(q.size() > 0)
        {
            int length = q.size();
            List<Integer> temp = new ArrayList<>();
            
            for(int i = 0; i < length; i++)
            {
                if(q.peek().left != null)
                {
                    q.offer(q.peek().left);
                }
                if(q.peek().right != null)
                {
                    q.offer(q.peek().right);
                }
                
                temp.add(q.poll().val);
            }
            ans.add(temp);
        }
        return ans;
    }
}

File: Aman/Tree/Binary Tree Level Order Traversal.cpp
class Solution {
public:
   
    vector<vector<int>> levelOrder(TreeNode* root) {
    
    queue<TreeNode*>q;
    vector<vector<int>>res;
    vector<int>v;    
        
    if(root==NULL)
        return res;
        
    q.push(root);
        
    while(!q.empty())
    {
       int n=q.size();
       for(int i=0;i<n;i++)
       {
           TreeNode* temp=q.front();
           q.pop();
           v.push_back(temp->val);
           if(temp->left)
             q.push(temp->left);
           if(temp->right)
             q.push(temp->right);
       }
       res.push_back(v);  
       v.resize(0);
    }
    return res;
    }
};
